src/transforms/transform-deployment.ts

```diff
@@ -139,31 +139,43 @@ export const mapEnvironment = (environment: string): string => {
 	return jiraEnv;
 };
 
-// Maps commit summaries to an array of a single association containing the commit keys (commit hash and repository id).
-// Returns undefined when there are no commit summaries to map.
-const mapCommitSummariesToAssociationArray = (
+// Maps issue ids and commit summaries to an array of associations (one for issue ids, and one for commits).
+// Returns undefined when there are no issue ids to map.
+const mapJiraIssueIdsAndCommitsToAssociationArray = (
+	issueIds: string[],
 	repositoryId: string,
 	commitSummaries?: CommitSummary[]
 ): JiraAssociation[] | undefined => {
 
-	if (!(commitSummaries && commitSummaries.length)) {
+	if (!(issueIds && issueIds.length)) {
 		return undefined;
 	}
 
-	const commitKeys = commitSummaries
-		.map((commitSummary) => {
-			return {
-				commitHash: commitSummary.sha,
-				repositoryId: repositoryId
-			};
-		});
-
-	return [
+	const associations: JiraAssociation[] = [
 		{
-			associationType: "commit",
-			values: commitKeys
+			associationType: "issueIdOrKeys",
+			values: issueIds
 		}
 	];
+
+	if (commitSummaries && commitSummaries.length) {
+		const commitKeys = commitSummaries
+			.slice(0, 500 - issueIds.length)
```

Comment:
What's with the magic `500` here?

Reply:
That’s the limit for all types of associations in an entity. I should have
added a constant and/or comment.